contracts/TalentLayerEscrow.sol

```diff
@@ -932,12 +967,29 @@ contract TalentLayerEscrow is
             transaction.token
         ] += originValidatedProposalFeeRate;
 
+        if (transaction.referrerId != 0 && transaction.referralAmount != 0) {
```

Comment:
to prevent any security issue, handle a referrerIdToTokenToBalance and add a new claim function